BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,41 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
-
-        var concatenation = Base58.Bitcoin.Decode(stringValue.AsSpan(3)).ToArray();
-        var hashedPublicKey = concatenation[..20];
-        var givenChecksum = concatenation[20..];
+        const string pattern = @"^did\:e\:(.+?)\:dids\:(.+)(.{2})$";
```

Comment:
```regex
^did\:e\:(.+?)\:dids\:(.+)(.{2})$
```

The `(.+?)` part is too open. We should 
* limit the number of valid characters to characters that are valid in a URL
* require a URL format

Reply:
> we're already limiting the length of the whole address. Wouldn't such a check be redundant?

I should have said the "set" of valid characters, instead of the "number" of valid characters.

Reply:
> we're already limiting the length of the whole address. Wouldn't such a check be redundant?

I should have said the "set" of valid characters, instead of the "number" of valid characters.